### ISSUE TITLE
False positive MixedPropertyFetch after `instanceof $this`

### ISSUE THREAD

**Issue by kynx** [vimeo/psalm#11281]
```php
<?php

abstract class A
{
    public function __construct(private string $value)
    {
    }

    public function equals(mixed $other): bool
    {
        if (! $other instanceof $this) {
            return false;
        }

        return $this->value === $other->value;
    }
}
```

Results in:

```
ERROR: MixedPropertyFetch - src/A.php:15:33 - Cannot fetch property on mixed var $other (see https://psalm.dev/034)
        return $this->value === $other->value;
```

**Comment by kynx** [comment]
https://psalm.dev/r/814236f6c7

**Comment by orklah** [comment]
Note that replacing $this by static works if it's not inverted
https://psalm.dev/r/0afbd90e20
